x/oracle/types/genesis.go

```diff
@@ -18,6 +22,13 @@ func DefaultGenesis() *GenesisState {
 // failure.
 func (gs GenesisState) Validate() error {
 	// this line is used by starport scaffolding # genesis/types/validate
-
+	if len(gs.Cyclelist) == 0 {
```

Comment:
an additional validation also could be to check if the queryType of the given queryData exists in x/registry and plus do this same cycleList validation in msgUpdateCyclist